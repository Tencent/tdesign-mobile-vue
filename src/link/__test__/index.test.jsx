import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import { Icon, AppIcon as TIconApp } from 'tdesign-icons-vue-next';
import Link from '../link';
// const iconFunc = () => h(TIconApp);

// every component needs four parts: props/events/slots/functions.
describe('Link', () => {
  // test props api
  describe('props', () => {
    it(':size', () => {
      const wrapper = mount({
        render() {
          return <Link size={'large'}>text</Link>;
        },
      });
      expect(wrapper.find('.t-link--large').exists()).toBeTruthy();
      const wrapperDefault = mount({
        render() {
          return <Link size={''}>text</Link>;
        },
      });
      expect(wrapperDefault.find('.t-link--medium').exists()).toBeTruthy();
    });
    it(':theme', () => {
      const wrapper = mount({
        render() {
          return <Link theme={'primary'}>text</Link>;
        },
      });
      expect(wrapper.find('.t-link--primary').exists()).toBeTruthy();
      const wrapperDefault = mount({
        render() {
          return <Link theme={''}>text</Link>;
        },
      });
      expect(wrapperDefault.find('.t-link--default').exists()).toBeTruthy();
    });
    it(':hover', () => {
      const wrapper = mount({
        render() {
          return <Link hover={true}>text</Link>;
        },
      });
      expect(wrapper.find('.t-link--hover').exists()).toBeTruthy();
    });
    it(':underline', () => {
      const wrapper = mount({
        render() {
          return <Link underline={true}>text</Link>;
        },
      });
      expect(wrapper.find('.t-link--underline').exists()).toBeTruthy();
    });
    it(':disabled', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return (
            <Link disabled={true} onClick={fn}>
              text
            </Link>
          );
        },
      });
      wrapper.trigger('click');
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper.find('.t-link--disabled').exists()).toBeTruthy();
    });
    it(':content', () => {
      const renderContent = function () {
        return 'foo';
      };
      const wrapper = mount({
        render() {
          return (
            <div>
              <Link content="foo">bar</Link>
              <Link content={renderContent}>bar</Link>
              <Link default="foo">bar</Link>
              <Link default={renderContent}>bar</Link>
              <Link content={'0'}>bar</Link>
            </div>
          );
        },
      });
      expect(wrapper.find('.t-link--content')).toBeTruthy();
    });
    it('prefixIcon', () => {
      const wrapper = mount(() => <Link prefixIcon={<Icon />}>text</Link>);
      expect(wrapper.findComponent(Icon)).toBeTruthy();
    });
  });

  // test events
  describe('@event', () => {
    it('Event passthrough ', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Link onClick={fn}>text</Link>;
        },
      });
      await wrapper.trigger('click');
      expect(wrapper.emitted('click')).toBeTruthy();
    });
  });

  // test slots
  describe('<slot>', () => {
    it(':icon', () => {
      const suffixIcon = () => <TIconApp />;
      const wrapper = mount(Link, {
        props: {
          suffixIcon,
        },
      });
      expect(wrapper.find('.t-link__suffix-icon').exists()).toBeTruthy();
    });
  });
});
