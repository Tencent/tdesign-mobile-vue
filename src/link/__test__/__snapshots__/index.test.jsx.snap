// Vitest Snapshot v1

exports[`Link > props > :content 1`] = `
<div>
  <a
    aria-disabled="false"
    class="t-link t-link--default t-link--medium"
    target=""
  >
    <!--v-if-->
    <span
      class="t-link__content"
    >
      
      bar
      
    </span>
    <!--v-if-->
  </a>
  <a
    aria-disabled="false"
    class="t-link t-link--default t-link--medium"
    target=""
  >
    <!--v-if-->
    <span
      class="t-link__content"
    >
      
      bar
      
    </span>
    <!--v-if-->
  </a>
  <a
    aria-disabled="false"
    class="t-link t-link--default t-link--medium"
    target=""
  >
    <!--v-if-->
    <span
      class="t-link__content"
    >
      
      bar
      
    </span>
    <!--v-if-->
  </a>
  <a
    aria-disabled="false"
    class="t-link t-link--default t-link--medium"
    target=""
  >
    <!--v-if-->
    <span
      class="t-link__content"
    >
      
      bar
      
    </span>
    <!--v-if-->
  </a>
  <a
    aria-disabled="false"
    class="t-link t-link--default t-link--medium"
    target=""
  >
    <!--v-if-->
    <span
      class="t-link__content"
    >
      
      bar
      
    </span>
    <!--v-if-->
  </a>
</div>
`;

exports[`Link > props > :disabled 1`] = `
<a
  aria-disabled="true"
  class="t-link t-link--default t-link--medium t-link--disabled"
  target=""
>
  <!--v-if-->
  <span
    class="t-link__content"
  >
    
    text
    
  </span>
  <!--v-if-->
</a>
`;

exports[`Link > props > :hover 1`] = `
<a
  aria-disabled="false"
  class="t-link t-link--default t-link--medium t-link--hover"
  target=""
>
  <!--v-if-->
  <span
    class="t-link__content"
  >
    
    text
    
  </span>
  <!--v-if-->
</a>
`;

exports[`Link > props > :size 1`] = `
<a
  aria-disabled="false"
  class="t-link t-link--default t-link--large"
  target=""
>
  <!--v-if-->
  <span
    class="t-link__content"
  >
    
    text
    
  </span>
  <!--v-if-->
</a>
`;

exports[`Link > props > :theme 1`] = `
<a
  aria-disabled="false"
  class="t-link t-link--primary t-link--medium"
  target=""
>
  <!--v-if-->
  <span
    class="t-link__content"
  >
    
    text
    
  </span>
  <!--v-if-->
</a>
`;

exports[`Link > props > :underline 1`] = `
<a
  aria-disabled="false"
  class="t-link t-link--default t-link--medium t-link--underline"
  target=""
>
  <!--v-if-->
  <span
    class="t-link__content"
  >
    
    text
    
  </span>
  <!--v-if-->
</a>
`;
